### PR TITLE
Refine terraform version constraint

### PR DESCRIPTION
**Pull request by ameukam**
Remove duplicate of terraform version constraint and ensure 0.14.x versions can't
be used.
Small improvements in verify-terraform.sh

Signed-off-by: Arnaud Meukam <ameukam@gmail.com>

### CODE REVIEW

File: infra/gcp/clusters/modules/gke-cluster/versions.tf [hunk 1]
```diff
@@ -15,7 +15,7 @@
  */
 
 terraform {
-  required_version = ">= 0.13"
+  required_version = ">= 0.13, < 0.14"
```
Comment (spiffxp) [inline infra/gcp/clusters/modules/gke-cluster/versions.tf:18]:
```suggestion
  required_version = "~> 0.13.0"
```
Ref https://www.terraform.io/docs/language/expressions/version-constraints.html

**Review by spiffxp — CHANGES_REQUESTED** [review]
Idea lgtm. Syntax nit

File: infra/gcp/clusters/modules/gke-cluster/versions.tf (hunk 1 above)
Comment (spiffxp) [reply infra/gcp/clusters/modules/gke-cluster/versions.tf:18]: ping @ameukam to fix this (across this and other files) when you find time

Comment (ameukam) [reply infra/gcp/clusters/modules/gke-cluster/versions.tf]: @spiffxp fix applied in the PR.

File: hack/verify-terraform.sh [hunk 2]
```diff
@@ -54,12 +55,12 @@ if [ -z "$(which curl)" ]; then
   exit 1
 fi
 
-
 if [ -z "$(which tfswitch)" ]; then
   echo "Installing tfswitch locally"
-  curl -L https://raw.githubusercontent.com/warrensbox/terraform-switcher/release/install.sh | sudo bash
+  curl -L https://raw.githubusercontent.com/warrensbox/terraform-switcher/release/install.sh | bash
```
Comment (spiffxp) [inline hack/verify-terraform.sh:60]: ❤️

**Review by spiffxp — APPROVED** [review]
/approve
/lgtm

**Comment by spiffxp** [comment]
Upgrading terraform ref: https://github.com/kubernetes/k8s.io/issues/2020

**Pull request merged**